tool/tctl/common/resource_command_test.go

```diff
@@ -1683,3 +1687,54 @@ version: v7
 	_, err = runResourceCommand(t, fc, []string{"create", "-f", roleYAMLPath})
 	require.NoError(t, err)
 }
+
+func testCreateServerInfo(t *testing.T, fc *config.FileConfig) {
+	// Ensure that our test server info does not exist
+	_, err := runResourceCommand(t, fc, []string{"get", types.KindServerInfo + "/test-server-info", "--format=json"})
+	require.True(t, trace.IsNotFound(err), "expected test-role to not exist prior to being created")
+
+	const serverInfoYAML = `kind: server_info
+sub_kind: cloud_info
+version: v1
+metadata:
+  name: test-server-info
+spec:
+  new_labels:
+    'a': '1'
+    'b': '2'
+`
+
+	// Create the server info
+	serverInfoYAMLPath := filepath.Join(t.TempDir(), "role.yaml")
+	require.NoError(t, os.WriteFile(serverInfoYAMLPath, []byte(serverInfoYAML), 0644))
```

Comment:
I'm not a fan of putting the code inside the test assertion. For me, this would read a lot better as

    err = os.WriteFile(serverInfoYAMLPath, []byte(serverInfoYAML), 0644)
    require.NoError(t, err)

It is much easier to scan down the left edge to see the code that is being run in the test. I know it is just setup code and not the code under test, but for tests I like to err on the side of clear and obvious even if it is more verbose that non-test code.

Reply:
My 2c, I'll sometimes do it like this, so the spacing draws attention to the method invocation (plus it sits in a separate line).

```go
assert.NoError(t,
    os.WriteFile(serverInfoYAMLPath, []byte(serverInfoYAML), 0644),
    "os.WriteFile(%q) errored", serverInfoYAMLPath)
```